agent/app/agent_unix.go

```diff
@@ -183,3 +187,27 @@ func contains(capabilities []string, capability string) bool {
 
 	return false
 }
+
+// initializeResourceFields exists mainly for testing doStart() to use mock Control
+// object
+func (agent *ecsAgent) initializeResourceFields() {
+	agent.resourceFields = &taskresource.ResourceFields{
+		Control: cgroup.New(),
+		IOUtil:  ioutilwrapper.NewIOUtil(),
+	}
+}
+
+func (agent *ecsAgent) cgroupInit() error {
```

Comment:
Is there unit test coverage for this, especially the `ExplicitlyEnabled` path?

Reply:
this is tested from parent method which is `doStart` for `ExplicitlyEnabled` for happy and error path. specifically [here](https://github.com/aws/amazon-ecs-agent/pull/1357/files/22aa850796dc440da28c32d264cd853fd622148b#diff-4e5ad8207820cdb7351cbdda04bfc94cR466).

Reply:
thanks for confirming!